contracts/src/SafeInternationalHarbour.sol

```diff
@@ -243,59 +117,33 @@ contract SafeInternationalHarbour {
             refundReceiver
         );
 
-        (address signer, bytes32 r, bytes32 vs) = _recoverSigner(
+        (address signer, bytes32 r, bytes32 vs) = CoreLib._recoverSigner(
             safeTxHash,
             signature
         );
 
-        // Store parameters only once (idempotent write)
-        SafeTransaction storage slot = _txDetails[safeTxHash];
-        if (!slot.stored) {
-            // first encounter → persist full parameter set
-            slot.stored = true;
-            slot.to = to;
-            slot.operation = operation;
-
-            // Writing to storage is expensive, so we only write if the value is non-zero
-            if (value > 0) {
-                slot.value = _safeCastUint256ToUint128(value);
-            }
-            if (safeTxGas > 0) {
-                slot.safeTxGas = _safeCastUint256ToUint128(safeTxGas);
-            }
-            if (baseGas > 0) {
-                slot.baseGas = _safeCastUint256ToUint128(baseGas);
-            }
-            if (gasPrice > 0) {
-                slot.gasPrice = _safeCastUint256ToUint128(gasPrice);
-            }
-            if (gasToken != address(0)) {
-                slot.gasToken = gasToken;
-            }
-            if (refundReceiver != address(0)) {
-                slot.refundReceiver = refundReceiver;
-            }
-            if (data.length > 0) {
-                slot.data = data;
-            }
-
-            emit NewTransaction(
-                safeTxHash,
-                safeAddress,
-                chainId,
-                nonce,
-                to,
-                value,
-                operation,
-                safeTxGas,
-                baseGas,
-                gasPrice,
-                gasToken,
-                refundReceiver,
-                data
-            );
-        }
+        _storeTransaction(
+            safeTxHash,
+            safeAddress,
+            chainId,
+            nonce,
+            to,
+            value,
+            data,
+            operation,
+            safeTxGas,
+            baseGas,
+            gasPrice,
+            gasToken,
+            refundReceiver
+        );
 
+        // --- DUPLICATE TRANSACTION SIGNATURE CHECK ---
```

Comment:
What is the rationale for moving this out of the `_storeSignature` function? I personally thought it fit well there.

Obviously not important for the PR, and I have no expectations to move it back. I was just curious.

Reply:
for me the name `_storeSignature` does not imply that a duplication check is done (and for `_storeTransaction` we don't). But more importantly for the 4337 version the duplication check and storage are done in separate flows (verification vs execution).